UnityClient/Assets/Scripts/Components/UI/Minimap/Minimap.cs

```diff
@@ -0,0 +1,296 @@
+using ROIO;
+using System;
+using System.IO;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class Minimap : MonoBehaviour
+{
+    [SerializeField] private Canvas _displayControl;
+    [SerializeField] private RectMask2D _mask;
+    [SerializeField] private RawImage _mapBase;
+    [SerializeField] private Button _btnPlus;
+    [SerializeField] private Button _btnMinus;
+    [SerializeField] private Button _btnMinimize;
+    [SerializeField] private TextMeshProUGUI _coordinates;
+    [SerializeField] private RawImage _playerIndicator;
+
+    [SerializeField] private int m_nZoom = DEFAULT_ZOOM_INDEX;
+    [SerializeField] private float m_fOffSetX = 0.0f;
+    [SerializeField] private float m_fOffSetY = 0.0f;
+    [SerializeField] private int m_nCenterX = 0;
+    [SerializeField] private int m_nCenterY = 0;
+
+    private Texture2D MapThumbTexture;
+    private Texture2D PlayerIndicatorTexture;
+
+    private const int DEFAULT_ZOOM_INDEX = 0;
+
+    private string currentMap;
+
+    private float m_fMapPicRealWidht = 0.0f;
+    private float m_fMapPicRealHeight = 0.0f;
+
+    private int m_nMapWidth;
+    private int m_nMapHeight;
+
+    private float m_f1MaskWidth = 0.0f;
+    private float m_f1MaskHeight = 0.0f;
+
+    private float[] m_fZoomValue =
+    {
+        0.25f,
+        0.32f,
+        0.425f,
+        0.75f,
+        1.0f
+    };
+
+    void Awake()
+    {
+        InitializeUI();
+
+        //GameEventUI.EventUpdateCurrentMiniMap += OnEventUpdateMap;
+        GameEventUI.EventUpdateCoordinateMiniMap += OnEventUpdateCoordinateMiniMap;
+    }
+
+    private void Start()
+    {
+        PlayerIndicatorTexture = FileManager.Load($"{DBManager.INTERFACE_PATH}map/map_arrow.bmp") as Texture2D;
+        _playerIndicator.texture = PlayerIndicatorTexture;
+        GameManager.OnMapLoaded += OnEventUpdateMap;
+    }
+
+    private void Update()
+    {
+        if(currentMap != null && _mapBase.texture == null)
+        {
+            OnEventUpdateMap();
+        }
+    }
+
+    void LateUpdate()
+    {
+        if (Session.CurrentSession == null)
+            return;
+
+        UpdatePlayerArrow();
+    }
+
+    void OnDestroy()
+    {
+        //GameEventUI.EventUpdateCurrentMiniMap -= OnEventUpdateMap;
+        GameEventUI.EventUpdateCoordinateMiniMap -= OnEventUpdateCoordinateMiniMap;
+        GameManager.OnMapLoaded -= OnEventUpdateMap;
+    }
+
+    void InitializeUI()
+    {
+        _btnPlus.onClick.RemoveAllListeners();
+        _btnPlus.onClick.AddListener(OnClickMiniMapPlus);
+        _btnMinus.onClick.RemoveAllListeners();
+        _btnMinus.onClick.AddListener(OnClickMiniMapMinus);
+
+        /*
+        Debug.Log("PASSEI AQUI");
+        var arrow = FileManager.Load($"{DBManager.INTERFACE_PATH}map/map_arrow.bmp") as Texture2D;
+        _playerIndicator.texture = arrow;
+        Debug.Log("PLAYER INDICATOR SET");
+        */
+    }
+
+    private void OnEventUpdateMap()
```

Comment:
you can change the `GameManager` event to pass the map name as well, make it simpler here